File: app/src/main/java/com/happs/ximand/ringcontrol/model/object/command/simple/SimpleBluetoothCommand.java
package com.happs.ximand.ringcontrol.model.object.command.simple;

import com.happs.ximand.ringcontrol.model.object.command.BluetoothCommand;

public abstract class SimpleBluetoothCommand extends BluetoothCommand<Byte> {

    private byte value;

    protected SimpleBluetoothCommand(byte code, byte value) {
        super(code);
        this.value = value;
    }

    @Override
    public byte[] getCommand() {
        return new byte[]{getCode(), value};
    }

    @Override
    public Byte getMainContent() {
        return value;
    }
}

File: app/src/main/java/com/happs/ximand/ringcontrol/model/object/command/simple/SimpleBluetoothCommand.kt
package com.happs.ximand.ringcontrol.model.`object`.command.simple

import com.happs.ximand.ringcontrol.model.`object`.command.BluetoothCommand

abstract class SimpleBluetoothCommand(code: Byte, val mainContent: Byte) : BluetoothCommand<Byte>(code) {

    override fun toByteArray(): ByteArray {
        return byteArrayOf(code, mainContent)
    }

    override fun getMainContent(): Byte {
        return mainContent
    }
}